controllers/component_controller.go

```diff
@@ -401,7 +403,7 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		if devfileLocation != "" {
 			// Parse the Component Devfile
 			log.Info(fmt.Sprintf("Parsing Devfile from the Devfile location %s... %v", devfileLocation, req.NamespacedName))
-			compDevfileData, err = cdqanalysis.ParseDevfileWithParserArgs(&devfileParser.ParserArgs{URL: devfileLocation, Token: gitToken})
+			compDevfileData, err = cdqanalysis.ParseDevfileWithParserArgs(&devfileParser.ParserArgs{URL: devfileLocation, Token: gitToken, DevfileUtilsClient: r.DevfileUtilsClient})
```

Comment:
I saw in component controller, it calls multiple devfile parser. why only pass in `DevfileUtilsClient` in this place? because the test case only covers here?
also should this be tested in CDQ module as well?

Reply:
Good Question. Because:
1. This is the only place in component controller which uses URL `devfileParser.ParserArgs{URL: devfileLocation ...`, other invocations use `&devfileParser.ParserArgs{Data: []byte(component.Status.Devfile) ....`
2. https://issues.redhat.com/browse/DEVHAS-492 deals with testing private links in the devfile, so when a Devfile is from a data src and has private links for URI for example, then we can provide `DevfileUtilsClient` in these calls
3. Also, devfile/library does a default assignment if its nil, https://github.com/devfile/library/blob/main/pkg/devfile/parser/parse.go#L173-L175. So even if we dont provide argument with the call here, it will always use the real implementation. When we are working on https://issues.redhat.com/browse/DEVHAS-492, we can change it to allow for mock tests.